### PR TITLE
Add crash reporting via signal handlers + disk persistence

### PR DESCRIPTION
**Pull request by moona3k**
## Context

The v0.4.22 onboarding incident stranded ~23 new users on the setup screen for ~24 hours — a broken Continue button meant they couldn't complete onboarding. While investigating via telemetry, we noticed some sessions showed zero activity after onboarding, raising the question: did they crash? We had no way to tell. The app had zero crash visibility — if the process dies, the telemetry service dies with it.

This PR closes that blind spot.

## Architecture

Same 3-step pattern used by Sentry, PLCrashReporter, and KSCrash — distilled to its minimum for a small indie macOS app:

```
1. install()        → Register signal handlers + ObjC exception handler (before NSApplication.run())
2. Signal fires     → Write crash report to disk via POSIX I/O (pre-allocated buffers, no heap)
3. Next launch      → Read crash file → send crash_occurred telemetry event → delete file
```

**Signals caught:** SIGSEGV, SIGABRT, SIGBUS, SIGILL, SIGTRAP, SIGFPE, plus ObjC uncaught exceptions.

**Crash report format** (plain text, line-oriented — not JSON, because balanced braces are risky in a signal handler):
```
crash_type: signal
signal: 11
name: SIGSEGV
timestamp: 1711900000
app_ver: 0.5.1
os_ver: 15.3.1
uuid: A1B2C3D4-E5F6-7890-ABCD-EF1234567890
slide: 0x100000
--- stack ---
0x00000001a2f3b4c0
0x00000001a2f3b4d8
```

**Symbolication:** Captures Mach-O UUID + ASLR slide, enabling offline symbolication via `atos -o MacParakeet -arch arm64 -l <slide> <addresses>`.

## Signal handler safety

The signal handler uses only async-signal-safe POSIX functions (`open`, `write`, `close`, `vsnprintf`, `backtrace`, `raise`, `time`). Pre-allocated static buffers avoid heap allocation. `OSAtomicCompareAndSwap32Barrier` prevents concurrent handler entry. `SA_RESETHAND | SA_NODEFER` ensures the handler falls through to the default if it itself crashes. `sigaltstack` provides an alternate stack for stack overflow crashes.

**Known tradeoff:** Writing signal handlers in Swift (rather than C) means some Swift runtime calls (`withUnsafeMutableBufferPointer`, `CVarArg` variadics) are not strictly async-signal-safe. In practice, the risk is limited to heap corruption crashes where `malloc` could deadlock — a minority of crash types, and those are also caught by Apple's out-of-process `ReportCrash`. See [review discussion](https://github.com/moona3k/macparakeet/pull/38#discussion_r3019549223) for the full analysis.

## Known limitations (documented, accepted)

- **`backtrace()` not strictly async-signal-safe** — can deadlock on dyld lock. Same tradeoff all major crash reporters make. Worst case: one report lost.
- **`SIGKILL` (OOM kills) cannot be caught** — fundamental OS limitation.
- **Swift async backtraces not captured** — only physical thread stack.
- **Single crash file** — if the app crashes twice before a successful launch, only the second crash is reported.

## Files changed

| File | Change |
|------|--------|
| `CrashReporter.swift` | New — signal handlers, ObjC exception handler, Mach-O UUID extraction, file parser, telemetry sender |
| `TelemetryEvent.swift` | Add `crashOccurred` event with crash metadata props |
| `TelemetryService.swift` | Add `crashOccurred` to immediate flush events |
| `MacParakeetApp.swift` | `CrashReporter.install()` as first line of `main()` |
| `AppEnvironment.swift` | `sendPendingReport(via:)` after telemetry is configured |
| `CrashReporterTests.swift` | 17 tests covering parsing, edge cases, telemetry integration |
| `TelemetryServiceTests.swift` | Add crash event to contract coverage |

## Test plan

- [x] 996 tests passing (983 XCTest + 13 Swift Testing), 0 failures
- [x] 17 CrashReporter tests: signal/exception parsing, missing/empty/malformed files, colons in values, non-hex stack lines, 256-frame cap, trailing whitespace, escaped newlines round-trip, telemetry integration (event sent + file deleted), disabled telemetry, no crash file
- [x] Plan reviewed by Gemini + Codex before implementation
- [x] Implementation reviewed by 13+ independent AI agents; all findings addressed
- [x] Backend `crash_occurred` event added to Cloudflare Worker allowlist and deployed

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: Sources/MacParakeetCore/Services/CrashReporter.swift [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+    }
+
+    // MARK: - Signal Handler (@convention(c), async-signal-safe)
+    //
+    // Uses only async-signal-safe functions per Darwin's man sigaction:
+    // snprintf, open, write, close, backtrace, raise, sigaction, time.
+
+    private static let signalHandler: @convention(c) (Int32) -> Void = { sig in
+        // Guard: only first thread proceeds
+        guard OSAtomicCompareAndSwap32Barrier(0, 1, &handlerEntered) else { return }
```
[the remark this answers is not in the document]
Comment (moona3k) [reply Sources/MacParakeetCore/Services/CrashReporter.swift]: Acknowledged. `OSAtomicCompareAndSwap32Barrier` is deprecated in favor of `stdatomic.h` / `os_atomic`, but neither is bridged to Swift. The deprecation is cosmetic — the function is stable ABI on Darwin and won't be removed. `os_atomic_cmpxchg` is a macro (not callable from Swift), and Swift's `Synchronization.Atomic` requires Swift 6's ownership model which adds complexity in a `@convention(c)` context.

If we later move the signal handler to a C bridging file (as discussed above), we'd use `atomic_compare_exchange_strong` from `stdatomic.h` at that point. For now, the deprecated API is the correct pragmatic choice.

File: Sources/MacParakeetCore/Services/CrashReporter.swift [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+    public static func install() {
+        guard !installed else { return }
+        installed = true
+
+        // 1. Ensure the crash directory exists
+        let dir = AppPaths.appSupportDir
+        var isDir: ObjCBool = false
+        if !FileManager.default.fileExists(atPath: dir, isDirectory: &isDir) {
+            try? FileManager.default.createDirectory(atPath: dir, withIntermediateDirectories: true)
+        }
```
[the remark this answers is not in the document]
Comment (moona3k) [reply Sources/MacParakeetCore/Services/CrashReporter.swift:71]: Valid. Fixed in follow-up PR #153 by handling the exists-but-not-directory app support path before crash writes.

File: Sources/MacParakeetCore/Services/CrashReporter.swift [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+            crashOsVer: report.osVersion,
+            uuid: report.uuid,
+            slide: report.slide,
+            reason: report.reason,
+            stackTrace: stackTraceString
+        ))
+
+        // Always delete — even if telemetry is disabled (send() handles opt-out)
+        deleteCrashFile(at: path)
+    }
```
[the remark this answers is not in the document]
Comment (moona3k) [reply Sources/MacParakeetCore/Services/CrashReporter.swift:393]: Valid. Fixed in follow-up PR #153 by making pending crash report sending async and deleting the crash file only after the telemetry event is flushed or intentionally dropped.
